graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,207 @@
+            "euclidean",
+            "l2",
+            "l1",
+            "manhattan",
+            "cityblock",
+            "braycurtis",
+            "canberra",
+            "chebyshev",
+            "correlation",
+            "cosine",
+            "dice",
+            "hamming",
+            "jaccard",
+            "kulsinski",
+            "mahalanobis",
+            "matching",
+            "minkowski",
+            "rogerstanimoto",
+            "russellrao",
+            "seuclidean",
+            "sokalmichener",
+            "sokalsneath",
+            "sqeuclidean",
+            "yule",
+            "wminkowski",
+            "nan_euclidean",
+            "haversine",
+            "gaussian",
+        ]
+
+        if not isinstance(test, str):
+            msg = "test must be a str, not {}".format(type(test))
+            raise TypeError(msg)
+        elif test not in _VALID_TESTS:
+            msg = "Unknown test {}. Valid tests are {}".format(test, _VALID_TESTS)
+            raise ValueError(msg)
+
+        if not isinstance(metric, str) and not callable(metric):
+            msg = "Metric must be str or callable, not {}".format(type(metric))
+            raise TypeError(msg)
+        elif metric not in _VALID_METRICS and not callable(metric):
+            msg = "Unknown metric {}. Valid metrics are {}, or a callable".format(
+                metric, _VALID_METRICS
+            )
+            raise ValueError(msg)
+
         if n_components is not None:
             if not isinstance(n_components, int):
-                msg = "n_components must an int, not {}.".format(type(n_components))
+                msg = "n_components must be an int, not {}.".format(type(n_components))
                 raise TypeError(msg)
 
         if not isinstance(n_bootstraps, int):
-            msg = "n_bootstraps must an int, not {}".format(type(n_bootstraps))
+            msg = "n_bootstraps must be an int, not {}".format(type(n_bootstraps))
             raise TypeError(msg)
         elif n_bootstraps < 1:
             msg = "{} is invalid number of bootstraps, must be greater than 1"
             raise ValueError(msg.format(n_bootstraps))
 
-        if bandwidth is not None and not isinstance(bandwidth, float):
-            msg = "bandwidth must an int, not {}".format(type(bandwidth))
+        if not isinstance(num_workers, int):
+            msg = "num_workers must be an int, not {}".format(type(num_workers))
             raise TypeError(msg)
+        elif num_workers <= 0:
+            msg = "{} is invalid number of workers, must be greater than 0"
+            raise ValueError(msg.format(num_workers))
+        elif num_workers > 1:
+            raise NotImplementedError()  # TODO env error parallelizing
 
         super().__init__(embedding="ase", n_components=n_components)
+
+        hyppo_input = ["CCA", "Dcorr", "HHG", "RV", "Hsic", "MGC"]
+        conv_dict = dict(zip(_VALID_TESTS, hyppo_input))
+        test = conv_dict[test]
```

Comment:
hyppo says they are not case sensitive so I don't think this matters

Reply:
@bdpedigo verified. it does not. will remove in a sec.

Reply:
fixed.